src/network/networkd-link.c

```diff
@@ -2481,10 +2449,14 @@ static int link_configure(Link *link) {
         if (r < 0)
                return r;
 
-        r = link_set_ipv6_proxy_ndp(link);
+        r = ipv6_proxy_ndp_set(link);
         if (r < 0)
                return r;
 
+        r = ipv6_proxy_ndp_addresses_configure(link);
```

Comment:
This goes in row with implying `IPForward` in case `IPMasquerade=yes` is set. I added a new commit on top which enables proxy_ndp in case at least one address is configured.